validator/gke_cluster_version_test.rego

```diff
@@ -0,0 +1,479 @@
+#
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+package templates.gcp.GKEClusterVersionConstraintV1
+
+# Importing the test data
+import data.test.fixtures.gke_cluster_version.assets as fixture_assets
+
+# Importing the test constraints for testing master version
+import data.test.fixtures.gke_cluster_version.constraints.master_version_allowlist_all as master_version_allowlist_all
+import data.test.fixtures.gke_cluster_version.constraints.master_version_allowlist_none as master_version_allowlist_none
+import data.test.fixtures.gke_cluster_version.constraints.master_version_allowlist_one as master_version_allowlist_one
+import data.test.fixtures.gke_cluster_version.constraints.master_version_allowlist_one_exemption as master_version_allowlist_one_exemption
+import data.test.fixtures.gke_cluster_version.constraints.master_version_denylist_all as master_version_denylist_all
+import data.test.fixtures.gke_cluster_version.constraints.master_version_denylist_none as master_version_denylist_none
+import data.test.fixtures.gke_cluster_version.constraints.master_version_denylist_one as master_version_denylist_one
+import data.test.fixtures.gke_cluster_version.constraints.master_version_denylist_one_exemption as master_version_denylist_one_exemption
+
+# Importing the test constraints for testing node version
+import data.test.fixtures.gke_cluster_version.constraints.node_version_allowlist_all as node_version_allowlist_all
+import data.test.fixtures.gke_cluster_version.constraints.node_version_allowlist_none as node_version_allowlist_none
+import data.test.fixtures.gke_cluster_version.constraints.node_version_allowlist_one as node_version_allowlist_one
+import data.test.fixtures.gke_cluster_version.constraints.node_version_allowlist_one_exemption as node_version_allowlist_one_exemption
+import data.test.fixtures.gke_cluster_version.constraints.node_version_denylist_all as node_version_denylist_all
+import data.test.fixtures.gke_cluster_version.constraints.node_version_denylist_none as node_version_denylist_none
+import data.test.fixtures.gke_cluster_version.constraints.node_version_denylist_one as node_version_denylist_one
+import data.test.fixtures.gke_cluster_version.constraints.node_version_denylist_one_exemption as node_version_denylist_one_exemption
+
+# Find all violations on our test cases
```

Comment:
I think this should use the new test utils.

Reply:
Ack - PTAL